pyfar/dsp/dsp.py

```diff
@@ -1487,4 +1494,354 @@ def decibel(signal, domain='freq', log_prefix=None, log_reference=1):
+              channel_handling='each', target=1, limits=(None, None),
+              unit=None, return_reference=False):
+    """
+    Apply a normalization.
+
+    In the default case, the normalization ensures that the maximum absolute
+    amplitude of the signal after normalization is 1. This is achieved by
+    the multiplication
+
+    ``signal_normalized = signal * target / reference``,
+
+    where `target` equals
+    1 and `reference` is the maximum absolute amplitude before the
+    normalization.
+
+    Several normalizations are possible, which in fact are different ways
+    of computing the `reference` value (e.g. based on the spectrum).
+    See the parameters for details.
+
+    Parameters
+    ----------
+    signal: Signal, TimeData, FrequencyData
+        Input signal.
+    domain: string
+        Determines whih data is used to compute the `reference` value.
+
+        ``'time'``
+           Use the time data ``signal.time``.
+        ``'freq'``
+          Use the frequency data ``signal.freq``. Note that the data with
+          FFT normalization is used (cf.
+          :py:mod:`FFT concepts <pyfar._concepts.fft>`).
+
+        The default is ``'time'``.
+    operation: string, optional
+        Operation to compute the channel-wise `reference` value using the data
+        according to `domain`.
+
+        ``'max'``
+            Compute the maximum absolute value per channel.
+        ``'mean'``
+            Compute the mean absolute values per channel.
+        ``'energy'``
+            Compute the energy per channel using :py:func:`~pyfar.dsp.energy`.
+        ``'power'``
+            Compute the power per channel using :py:func:`~pyfar.dsp.power`.
+        ``'rms'``
+            Compute the RMS per channel using :py:func:`~pyfar.dsp.rms`.
+
+        The default is ``'max'``.
+    channel_handling: string, optional
+        Define how channel-wise `reference` values are handeled for multi-
+        channel signals. This parameter does not affect single-channel signals.
+
+        ``'each'``
+            Separate normalization of each channel.
+        ``'max'``
+            Normalize to the maximum `reference` value across channels.
+        ``'min'``
+            Normalize to the minimum `reference` value across channels.
+        ``'mean'``
+            Normalize to the mean `reference` value across the channels.
+
+       The default is ``'each'``.
+    target: scalar, array
+        The target to which the signal is normalized. Can be a scalar or an
+        array. In the latter case the shape of `target` must be broadcastable
+        to ``signal.cshape``. The default is ``1``.
+    limits: tuple, array_like
+        Restrict the time or frequency range that is used to compute the
+        `reference`value. Two element tuple specifying upper and lower limit
+        according to `domain` and `unit`. A `None` element means no upper or
+        lower limitation. The default ``(None, None)`` uses the entire signal.
```

Comment:
Do we want to add the information, that in case of samples/bin the second value defines the first sample/bin that is excluded?